src/components/UserController.tsx

```diff
@@ -1,22 +1,34 @@
 import React, { useState } from "react";
+import { useDispatch } from "redux-react-hook";
 import { useUserContext } from "../customHooks/useUserContext";
-import { User } from "../types";
 import { putUser } from "../lib/backend";
+import { updateUser } from "../store/reducers/user";
 
+export const UserController = () => {
+  const user = useUserContext()!;
 
-const saveUser = (displayName: string, user: User) => {
-  putUser({ ...user, displayName });
-}
+  const dispatch = useDispatch();
 
-export const UserController = () => {
-  const user = useUserContext();
   const [displayName, setDisplayName] = useState(user.displayName);
+
+  const saveUser = () => {
+    putUser({ ...user, displayName }).then(user => {
```

Comment:
Rename inner `user` to `updatedUser` to avoid confusion?

Reply:
You got it!